File: lib/data/album.dart
import 'package:media_picker_builder/data/media_file.dart';

class Album {
  /// Unique identifier for the album
  final String id;
  final String name;
  final List<MediaFile> files;

  Album({this.id, this.name, this.files});

  Album.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        files = json['files']
            .map<MediaFile>((json) => MediaFile.fromJson(json))
            .toList();

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Album && runtimeType == other.runtimeType && id == other.id;

  @override
  int get hashCode => id.hashCode;
}

File: android/src/main/kotlin/com/kasem/media_picker_builder/Album.kt
package com.kasem.media_picker_builder

import org.json.JSONArray
import org.json.JSONObject

class Album(
        var id: Long,
        var name: String,
        var files: MutableList<MediaFile>) {


    fun toJSONObject(): JSONObject {
        return JSONObject()
                .put("id", id.toString())
                .put("name", name)
                .put("files", JSONArray(files.map { it.toJSONObject() }))
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Album

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }


}